src/core/api/nat64_api.cpp

```diff
@@ -66,6 +66,33 @@ void otNat64SetReceiveIp4Callback(otInstance *aInstance, otNat64ReceiveIp4Callba
 {
     AsCoreType(aInstance).Get<Ip6::Ip6>().SetNat64ReceiveIp4DatagramCallback(aCallback, aContext);
 }
+
+otError otNat64GetNextAddressMapping(otInstance *                   aInstance,
+                                     otNat64AddressMappingIterator *aIterator,
+                                     otNat64AddressMapping *        aMapping)
+{
+    return AsCoreType(aInstance).Get<Nat64::Translator>().GetNextAddressMapping(*aIterator, *aMapping);
+}
+
+void otNat64GetCounters(otInstance *aInstance, otNat64ProtocolCounters *aCounters)
+{
+    AsCoreType(aInstance).Get<Nat64::Translator>().GetCounters(*aCounters);
```

Comment:
Same here?
If we define a related core type and then use `AsCoreType()` on `aCounters`,  it takes care of the non-null assert for us